### DISCUSSION TITLE
Stop unwanted webhook

### DISCUSSION THREAD

**Discussion by Icallhimtest** [community/community#50817] [Apps, API and Webhooks]
### Select Topic Area

Product Feedback

### Body

**tl;dr:**

As the payload URL of a Webhook, I'd like to tell Github to stop sending the Webhook deliveries.
(By returning a "410 Gone" HTTP code for example.)

**Long version**

We have a GitHub app where we create webhooks for our customer logged in through GitHub OAuth. After a certain time of inactivity we delete the customer's data but we have no way of removing their webhooks for them as we don't have access to their repositories if they're not connected. 

-> We end up spammed with webhooks we don't need and Github keeps sending out useless deliveries.

It would be great if there was a way to tell Github upon delivery that the Webhook can be removed. One way would be to have GitHub remove the webhook upon receiving a "410 Gone" HTTP status code for one of its deliveries.

Similar topic that doesn't ask this explicitly: https://github.com/orgs/community/discussions/30579

PS: I had a topic opened on the previous community forums about 3 years ago that had half a dozen of endorsements by other users requiring this as well (broken link: https://github.community/t/delete-a-webhook-from-unwanted-repository/14124)
